scripts/listTopics.js

```diff
@@ -9,40 +9,52 @@ const TOPIC_COUNT = 0;
  * @param  {object} azureServiceBus - Service Bus Instance
  * @param  {number} numberOfTopics - Number of topics to get
  * @param  {number} skip - Number of topics to skip
+ * @param  {number} startsWith - Topics that start with this
  * @param  {number} count - Topic count to start at
  * @param  {function} onDone - Invoked when we have built all the topics
  * @param  {function} onError - Invoked on error
  * @param  {array} topics - Array to store list of topics
  */
-function getNext(azureServiceBus, numberOfTopics, skip, count, onDone, onError, topics = []) {
-  let topicCount = count;
+function getNext(azureServiceBus, numberOfTopics, skip, startsWith, count, onDone, onError, topics = []) {
+    let topicCount = count;
 
-  azureServiceBus.listTopics({ top: numberOfTopics, skip }, (error, result) => {
-    if (error) return onError(error);
+    if (!numberOfTopics || numberOfTopics === 0) {
+        numberOfTopics = NUMBER_OF_TOPICS;
+    }
 
-    result.forEach((topic) => {
-      topics.push(topic.TopicName);
-      topicCount += 1;
-    });
+    azureServiceBus.listTopics({top: numberOfTopics, skip}, (error, result) => {
+        if (error) return onError(error);
 
-    if (result.length === numberOfTopics) {
-      getNext(azureServiceBus, numberOfTopics, skip + numberOfTopics, topicCount, onDone, onError, topics);
-    } else {
-      return onDone({ topics, topicCount });
-    }
-  });
+        result.forEach((topic) => {
+            if ((startsWith && startsWith.length > 0) && !topic.TopicName.startsWith(startsWith)) {
+                return;
+            }
+            topics.push(topic.TopicName);
+            topicCount += 1;
+        });
+
+        if (result.length > 0 && topics.length !== numberOfTopics) {
+            let numberToSkip = +skip + +numberOfTopics;
```

Comment:
I don't think this line is necessary. Recommend keeping this the same as it was in the original (inline addition).

Reply:
for some reason it was concatenating as a string thus the reason for + sign before the variable.